internal/hash/reader.go

```diff
@@ -257,6 +257,24 @@ func (r *Reader) Read(p []byte) (int, error) {
 		r.contentHasher.Write(p[:n])
 	}
 
+	// If we have reached our expected size,
```

Comment:
I am not sure if this is really needed, but it is a nice safety net if downstream should add an `io.LimitedReader` that will not hit EOF.

Checksums are relying on hitting EOF to be verified.